src/cnab.rs

```diff
@@ -69,24 +67,64 @@ impl Bundle {
     /// let bundle = Bundle::from_file("testdata/bundle.json").unwrap();
     /// assert_eq!(bundle.name, "helloworld");
     /// ```
-    pub fn from_file<P: AsRef<Path>>(path: P) -> Result<Self, BundleParseError> {
-        let file = File::open(path)?;
-        Self::from_json(file)
-    }
-
-    /// Deserialize a `Bundle` from any type implementing `Read`.
-    pub fn from_json<R: Read>(reader: R) -> Result<Self, BundleParseError> {
-        let bundle = serde_json::from_reader(reader)?;
-        Ok(bundle)
+    pub fn from_file(file_path: &str) -> Result<Self, BundleParseError> {
+        let file = File::open(Path::new(&file_path))?;
+        let res: Result<Bundle, canonical_json::Error> = canonical_json::from_reader(file);
+        match res {
+            Ok(b) => Ok(b),
+            Err(err) => {
+                match err {
+                    // the canonical JSON parser errors out if the input JSON is not canonical
+                    // for now, we should accept non-canonical JSON as input
+                    canonical_json::Error::Syntax(
+                        canonical_json::SyntaxError::UnexpectedWhitespace,
+                        _,
+                        _,
+                    ) => {
+                        let file = File::open(Path::new(&file_path)).expect("file not found");
```

Comment:
Q: Why are you opening the file again? I think what you want is to rewind the file handle to the beginning of the file using seek:

https://doc.rust-lang.org/std/io/trait.Seek.html